app-misc/bb/files/bb-1.3.0_rc1-fix-build-for-clang16.patch

```diff
@@ -0,0 +1,23 @@
+The two functions that are assigned to the formulas structs secnod field
+`int (*calculate)(number_t number_t number_t number_t)`
+(without the REGISTERS(3) which is just a macro to GCC regparm)
+
+However, clang16 has -Wincompatible-function-pointer-types by default, and it
+does not like that and complains, that assigning these functions into the
+structs fields is a problem due to incompatible types. 
+
+Bug: https://bugs.gentoo.org/880385
+
+Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>
+
+--- a/formulas.h
++++ b/formulas.h
+@@ -50,7 +50,7 @@ struct symetryinfo {
+ #define FORMULAMAGIC 1121
+ struct formula {
+     int magic;
+-    int (*calculate) (number_t, number_t, number_t, number_t) REGISTERS(3);
```

Comment:
I think it is okay to remove regparm here (REGISTERS is just a macro to that) isn't it? 
I mean, this is from 1997. I think most computers should be able to render some ascii art without using cpu registers unless someone is running this on a calculator.

Reply:
`register` as a keyword is complete snakeoil, regparm is a bit different (consider it like forced inlining, it's probably not necessary) but I think it can still go.

This might be a problem to just drop if asm is used anywhere in the program. I see it used in ctrl87.c but not sure if `formula` gets passed through to there or not.

Reply:
This is a Demo for an ASCII library. I think it is save to just have try to compile it with the patch and then watch the demo. If it still works, it works.